src/components/layout/Undernames/Undernames.tsx

```diff
@@ -81,12 +116,100 @@ function Undernames() {
     selectedUndernameRow,
     undernameTableLoading,
     percentUndernamesLoaded,
+    undernameAction,
   ]);
 
+  function resetActionModal() {
+    setUndername(undefined);
+    setTargetID(undefined);
+    setTTL(undefined);
+    setAction(undefined);
+    setSelectedRow(undefined);
+  }
+  async function handleOnNext() {
+    try {
+      if (!id) {
+        throw new Error('No PDNT ID found, unable to perform transaction.');
+      }
+      const confirmations = await arweaveDataProvider.getTransactionStatus(
+        new ArweaveTransactionID(id),
+      );
+      if (confirmations < 1) {
+        throw new Error(
+          'PDNT must have a minimum of 1 confirmation before editing, this protects against editing contracts lost due to block reorgs.',
+        );
+      }
+      switch (action) {
+        case UNDERNAME_TABLE_ACTIONS.CREATE:
```

Comment:
i'm assuming this is a switch bc other actions logic will be added here?

Reply:
yup, edit and remove